src/main/scala/com/fulcrumgenomics/util/Sequences.scala

```diff
@@ -55,25 +58,73 @@ object Sequences {
     * @param s a sequence
     * @return a tuple of (0-based index, length) of the longest homopolymer in the string
     */
+  @deprecated("Use maxPolyX instead", since="0.7.0")
   def longestHomopolymer(s: String): (Int, Int) = {
-    val xs = s.toUpperCase
-    val len = xs.length
-    assert(len > 0, "Cannot compute longest homopolymer of a zero length string")
+    assert(s.length > 0, "Cannot compute longest homopolymer of a zero length string")
+    val result = maxPolyX(s)
+    (result.offset, result.length)
+  }
+
+
+  case class OffsetAndLength(offset: Int, length: Int)
+
+  /**
+    * Returns the offset (0-based) and length of the longest homopolymer in the sequence. In the case
+    * that there are multiple homopolymers of the same length, the earliest one is returned.
+    *
+    * @param s a DNA or RNA sequence
+    * @return the offset and length of the longest homopolymer
+    */
+  def maxPolyX(s: String) : OffsetAndLength = {
+    val bs = s.getBytes
+    var (bestStart, bestLength) = (0,0)
+    var start = 0
+    while (start < bs.length) {
+      val firstBase = s.charAt(start).toByte
+      var i = start + 1
+      while (i < s.length && SequenceUtil.basesEqual(firstBase, bs(i))) i += 1
+      val length = i - start
+      if (length > bestLength) {
+        bestStart = start
+        bestLength = length
+      }
+      start = i
+    }
 
-    var foundIndex = 0
-    var foundLength = 1
+    OffsetAndLength(bestStart, bestLength)
+  }
 
-    forloop (from=0, until=len) { i =>
-      val x = xs.charAt(i)
-      var j = i+1
-      while (j < len && xs.charAt(j) == x) j += 1
-      val newLength = j - i
-      if (newLength > foundLength) {
-        foundIndex = i
-        foundLength = newLength
+  /**
+    * Returns the offset (0-based) and length of the longest dinucleotide run in the sequence. In the case
+    * that there are multiple dinucleotide runs of the same length, the earliest one is returned.
+    *
+    * @param s a DNA or RNA sequence
+    * @return the offset and length of the longest dinucleotide sequence
+    */
+  def maxDinuc(s: String) : OffsetAndLength = {
+    val bs = s.toUpperCase.getBytes
+    var (bestStart, bestLength) = (0,0)
+    forloop(0, bs.length-1) { start =>
+      val (b1, b2) = (bs(start), bs(start+1))
+      var i = start
+      while (i < s.length-1 && SequenceUtil.basesEqual(b1, bs(i)) && SequenceUtil.basesEqual(b2, bs(i+1))) i += 2
```

Comment:
Two thoughts:
1. If you upper case like you're doing on 105 you don't really need to use `SequenceUtil.basesEqual()` here.
2. If you continue to use `SequenceUtil.basesEqual()` I'd be tempted to define and use the following within this class:

```scala
/** Checks to see if the two characters are identical, ignoring case. */
@inline final private def same(a: Char, b: Char): Boolean = {
  Character.toUpperCase(a) == Character.toUpperCase(b)
}
```

This has the advantage that a) it'll work for all characters so if you try to use it on proteins it won't fail, and also will just make the usage more compact and easier to see what's going on.

Reply:
Done, but "I did it my way"